modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/cluster/programs/Joining.scala

```diff
@@ -149,29 +46,19 @@ sealed abstract class Joining[F[_]: Async: GenUUID: SecurityProvider: KryoSerial
   selfId: PeerId,
   stateAfterJoining: NodeState,
   versionHash: Hash,
-  joiningQueue: Queue[F, P2PContext]
+  peerDiscovery: PeerDiscovery[F]
 ) {
 
   private val logger = Slf4jLogger.getLogger[F]
 
   def join(toPeer: PeerToJoin): F[Unit] =
-    for {
-      _ <- validateJoinConditions()
-      _ <- session.createSession
-
-      _ <- joiningQueue.offer(toPeer)
-    } yield ()
+    validateJoinConditions() >>
+      session.createSession >>
+      Daemon.spawn(joinAll(toPeer)).start
```

Comment:
It's started like that so that it returns immediately (the call on the endpoint finishes)?

Reply:
Yes, we don't want to block the HTTP request. The request returns after starting the joining process.

Reply:
is it a supervised start? It's not obvious looking at `Daemon.spawn` 🤔